shed/event_streams.py

```diff
@@ -665,16 +683,29 @@ def __init__(self, predicate, child, *args, input_info,
         self.kwargs = kwargs
         self.args = args
         self.full_event = full_event
+        self.document_name = document_name
         self.generate_provenance(predicate=predicate)
 
+    def start(self, docs):
+        if self.document_name == 'start':
+            # TODO: should we have something like event_contents for starts?
+            if not self.predicate(docs):
+                # if we fail the criteria don't issue any documents
+                self.bypass = True
```

Comment:
Travis is failing (a set of documents is passing through without a `start`, which should not be allowed. Great idea to define `SinkAssertion` which caught it!). Looks like this `bypass` flag may not be doing what's intended (if set to `True`, then don't issue all docs involved with this `start`). I think that's where the error could come from.
Might be good to look this over. And maybe also to add a test for `bypass` (or a more rigorous one if tested).